boards/common/particle-mesh/bootloader.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2014-2016 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Christian Amsüss <chrysn@fsfe.org>
+ */
+
+#ifdef PARTICLE_MONOFIRMWARE
+#include <stdint.h>
+#include "vectors_cortexm.h"
+
+extern const void *_isr_vectors;
+
+/* As described in Particle's dynalib/inc/module_info.h (there under LGPL-3
```

Comment:
That one also does not use `__attribute__ ((packed))` to get rid of implicit padding?

Reply:
The upstream one has one, but it's really pointless -- the struct *is* already perfectly aligned and thus doesn't produce any padding (at least for 32bit and that's set in stone for this thing anyway).